packages/data-context/src/actions/LocalSettingsActions.ts

```diff
@@ -11,25 +11,34 @@ export interface LocalSettingsApiShape {
   setPreferences (object: AllowedState): Promise<void>
 }
 
+// Combine two objects into one either using Lodash merge strategy or the spread operator
+function combine (first: Object = {}, second: Object = {}, merge: boolean) {
```

Comment:
I added this because I needed to overwrite the `notifyWhenRunCompletes` property, but when we use `_.merge`, Lodash merges the list items together instead of overwriting the property.

Reply:
Could we use [`_.mergeWith`](https://lodash.com/docs/4.17.15#mergeWith) and provide a customizer to handle the array case? The non-merge case loses a lot of the recursive behavior we get with `merge`